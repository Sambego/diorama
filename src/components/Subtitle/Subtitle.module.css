.subtitle {
  margin: 0;
  max-width: 80%;
  font-size: 4.5vw;
  font-weight: 700;
  text-transform: uppercase;
}
