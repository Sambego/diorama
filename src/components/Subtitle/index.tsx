import React from "react";
import PropTypes from "prop-types";

import styles from "./Subtitle.module.css";

type SubtitleProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
}>;

function Subtitle({ children, style, className }: SubtitleProps) {
	return (
		<h2
			style={style}
			className={`${styles.subtitle} diorama-subtitle ${className}`}
		>
			{children}
		</h2>
	);
}

Subtitle.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	style: PropTypes.shape({}),
};

Subtitle.defaultProps = {
	className: "",
	style: {},
};

export default Subtitle;
