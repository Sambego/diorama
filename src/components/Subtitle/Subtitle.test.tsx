import React from "react";
import renderer from "react-test-renderer";
import Subtitle from ".";

it("renders correctly", () => {
	const tree = renderer.create(<Subtitle>Subtitle content</Subtitle>).toJSON();
	expect(tree).toMatchSnapshot();
});
