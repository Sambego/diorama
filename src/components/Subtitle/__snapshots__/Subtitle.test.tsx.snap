// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<h2
  className="_subtitle_1majr_1 diorama-subtitle "
  style={{}}
>
  Subtitle content
</h2>
`;
